wgpu-core/src/instance.rs

```diff
@@ -217,6 +218,35 @@ impl AdapterInfo {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+/// Error when requesting a device from the adaptor
+pub enum AdapterError {
+    /// Unsupported feature extension was requested
+    UnsupportedFeature(wgt::Features),
+    /// Adapter doesn't support the amount of requested bind groups
+    MaxBindGroupsExceeded { supported: u32, requested: u32 },
```

Comment:
I think this can be `LimitsExceeded`, as otherwise we'd have a lot of variants here, and we don't really care to return a super detailed error.

Reply:
Sure, may I suggest keeping the maximum supported bind groups by the adapter in the error? So you don't have to gradually decrease the requested number to guess the maximum supported by the hardware when developing. I feel that could be useful at least. But the requested number is redundant.